morphology/morph.py

```diff
@@ -0,0 +1,175 @@
+#! /usr/bin/python
+#
+# morph.py
+#
+# Usage:
+#     module for madlib.py
+#
+# Apply morphology to words
+#
+
+import sys
+import os
+import re
+
+# Irregular word .txt file dictionaries
+cwd = os.getcwd()
+nouns_filename = f"{cwd}/morphology/irreg_nouns.txt"
+verbs_filename = f"{cwd}/morphology/irreg_verbs.txt"
+artcls_filename = f"{cwd}/morphology/irreg_artcls.txt"
+
+# Open files and create python dictionaries
+irreg_nouns = {}
+try:
+    file_handle = open(nouns_filename, 'r')
+except:
+    print(f" ERROR: Failed to open words file '{nouns_filename}'")
+    print("  Exiting system from morphology/morph.py\n")
+    sys.exit()
+
+num_lines = 0
+while True:
+    num_lines += 1
+    line = file_handle.readline()
+    if not line:
+        # at end of file
+        break
+    # get rid of dashes
+    line = line.replace('–', '')
+    # split words in line
+    words = line.split()
+    irreg_nouns[words[0]] = words[1]
+#print(f"Irregular nouns: {irreg_nouns}\n")
+
+
+irreg_verbs = {}
+try:
+    file_handle = open(verbs_filename, 'r')
+except:
+    print(f" ERROR: Failed to open words file '{verbs_filename}'")
+    print("  Exiting system from morphology/morph.py\n")
+    sys.exit()
+
+num_lines = 0
+while True:
+    num_lines += 1
+    line = file_handle.readline()
+    if not line:
+        # at end of file
+        break
+    # get rid of dashes
+    line = line.replace('–', '')
+    # split words in line
+    words = line.split()
+    irreg_verbs[words[0]] = {'past':words[1], 'part':words[2], 'prog':words[3], 'pres':words[4]}
+#print(f"Irregular verbs: {irreg_verbs}\n")
+
+irreg_artcls = []
+try:
+    file_handle = open(artcls_filename, 'r')
+except:
+    print(f" ERROR: Failed to open words file '{artcls_filename}'")
+    print("  Exiting system from morphology/morph.py\n")
+    sys.exit()
+
+num_lines = 0
+while True:
+    num_lines += 1
+    line = file_handle.readline()
+    if not line:
+        # at end of file
+        break
+    irreg_artcls.append(line[:-1])
+#print(f"Words with irregular articles: {irreg_artcls}\n")
+
+#sys.exit()
+
+# Spelling rule patterns
+y_vowel = re.compile("[bcdfghjklmnpqrstvwxyz]y$")
+short_vowel = re.compile("[bcdfghjklmnpqrstvwxyz][aeiou][mpbntdg]$")
+
+# Morphology functions
+def pl(noun):
+    yv = y_vowel.findall(noun)
+    #print(yv)
+    if noun in irreg_nouns:
+        plural = irreg_nouns[noun]
+        return plural
+    elif yv:
+        plural = noun[:-1] + 'ies'
+        return plural
+    elif noun.endswith(('s', 'z', 'sh', 'ch', 'o')):
+        plural = noun + 'es'
+        return plural
+    else:
+        plural = noun + 's'
+        return plural
```

Comment:
Each section of the **if-elif-else** logic flow updates the `plural` variable.  No need to `return plural` in each context, instead just `return plural` once at the end --> cleaner code, fewer lines, and also slightly easier to read.